src/lib/location.js

```diff
@@ -0,0 +1,194 @@
+			{ name: 'iPad', value: 'iPad', version: 'OS' },
+			{ name: 'Kindle', value: 'Silk', version: 'Silk' },
+			{ name: 'Android', value: 'Android', version: 'Android' },
+			{ name: 'PlayBook', value: 'PlayBook', version: 'OS' },
+			{ name: 'BlackBerry', value: 'BlackBerry', version: '/' },
+			{ name: 'Macintosh', value: 'Mac', version: 'OS X' },
+			{ name: 'Linux', value: 'Linux', version: 'rv' },
+			{ name: 'Palm', value: 'Palm', version: 'PalmOS' },
+		],
+		databrowser: [
+			{ name: 'Chrome', value: 'Chrome', version: 'Chrome' },
+			{ name: 'Firefox', value: 'Firefox', version: 'Firefox' },
+			{ name: 'Safari', value: 'Safari', version: 'Version' },
+			{ name: 'Internet Explorer', value: 'MSIE', version: 'MSIE' },
+			{ name: 'Opera', value: 'Opera', version: 'Opera' },
+			{ name: 'BlackBerry', value: 'CLDC', version: 'CLDC' },
+			{ name: 'Mozilla', value: 'Mozilla', version: 'Mozilla' },
+		],
+		init() {
+			const agent = this.header.join(' ');
+			const os = this.matchItem(agent, this.dataos);
+			const browser = this.matchItem(agent, this.databrowser);
+
+			return { os, browser };
+		},
+		matchItem(string, data) {
+			let i = 0;
+			let j = 0;
+			let regex;
+			let regexv;
+			let match;
+			let matches;
+			let version;
+
+			for (i = 0; i < data.length; i += 1) {
+				regex = new RegExp(data[i].value, 'i');
+				match = regex.test(string);
+				if (match) {
+					regexv = new RegExp(`${ data[i].version }[- /:;]([\\d._]+)`, 'i');
+					matches = string.match(regexv);
+					version = '';
+					if (matches) { if (matches[1]) { matches = matches[1]; } }
+					if (matches) {
+						matches = matches.split(/[._]+/);
+						for (j = 0; j < matches.length; j += 1) {
+							if (j === 0) {
+								version += `${ matches[j] }.`;
+							} else {
+								version += matches[j];
+							}
+						}
+					} else {
+						version = '0';
+					}
+					return {
+						name: data[i].name,
+						version: parseFloat(version),
+					};
+				}
+			}
+			return { name: 'unknown', version: 0 };
+		},
+	};
+
+	const info = module.init();
+	return {
+		os: info.os.name,
+		osVersion: info.os.version,
+		browserName: info.browser.name,
+		browserVersion: info.browser.version,
+	};
+};
+
+/**
+ * This is used to convert location to a default type we want to send to server
+ * @param {Object} location
+ * @returns {Object}
+ */
+const convertLocationToSend = (location) => (
+	{
+		countryName: location.country || location.country_name,
+		countryCode: location.country_code,
+		city: location.city || location.state,
+		latitude: location.latitude,
+		longitude: location.longitude,
+	});
+
+/**
+ * This is used to get location details for user
+ * @param {Number} latitude
+ * @param {Number} longitude
+ * @returns {Object}
+ */
+const locationPrimary = async(latitude, longitude) => {
+	const { address } = await fetch(`https://nominatim.openstreetmap.org/reverse?format=json&lat=${ latitude }&lon=${ longitude }`, {
```

Comment:
So this is going out to this api to grab location?  This has to definitely be optional if done at all.  Some serious privacy concerns around sending data like this out.